codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RestJsonProtocolGenerator.java

```diff
@@ -44,55 +61,305 @@
+            MemberShape memberShape = payloadBinding.get().getMember();
+            Shape payloadShape = model.expectShape(memberShape.getTarget());
+            ShapeType shapeType = payloadShape.getType();
+            String memberName = symbolProvider.toMemberName(memberShape);
+
+            Optional<MediaTypeTrait> mediaTypeTrait = payloadShape.getTrait(MediaTypeTrait.class);
+            mediaTypeTrait.ifPresent(typeTrait -> writer.write("restEncoder.SetHeader(\"Content-Type\").String($S)",
+                    typeTrait.getValue()));
+
+            if (shapeType == ShapeType.BLOB) {
+                writer.write("documentPayload = input.$L", memberName);
+            } else if (shapeType == ShapeType.STRING) {
+                writer.write("documentPayload = []byte(input.$L)", memberName);
+            } else {
+                String functionName = ProtocolGenerator.getDocumentSerializerFunctionName(payloadShape,
+                        getProtocolName());
+                writer.write("jsonEncoder := json.NewEncoder()");
+                writer.openBlock("if err := $L(input.$L, jsonEncoder.Value); err != nil {", "}", functionName,
+                        memberName, () -> {
+                            writer.write("return err");
+                        });
+                writer.write("documentPayload = jsonEncoder.Bytes()");
+            }
+        } else {
+            writer.write("restEncoder.SetHeader(\"Content-Type\").String($S)", getDocumentContentType());
+            writer.write("");
+
+            Shape inputShape = model.expectShape(operation.getInput()
+                    .orElseThrow(() -> new CodegenException("input shape is missing on " + operation.getId())));
+            String functionName = ProtocolGenerator.getDocumentSerializerFunctionName(inputShape, getProtocolName());
+            writer.write("jsonEncoder := json.NewEncoder()");
+            writer.openBlock("if err := $L(input, jsonEncoder.Value); err != nil {", "}", functionName, () -> {
+                writer.write("return err");
+            });
+            writer.write("documentPayload = jsonEncoder.Bytes()");
+        }
+        writer.write("");
+
+        writer.openBlock("if request, err = request.SetStream(bytes.NewReader(documentPayload)); err != nil {", "}",
+                () -> {
+                    writer.write("return out, metadata, &aws.SerializationError{Err: err}");
+                });
     }
 
     @Override
-    protected void writeErrorCodeParser(GenerationContext context) {
-        // pass
+    protected void generateDocumentBodyShapeSerializers(GenerationContext context, Set<Shape> shapes) {
+        GoWriter writer = context.getWriter();
+        Model model = context.getModel();
+        SymbolProvider symbolProvider = context.getSymbolProvider();
+
+        shapes.forEach(shape -> {
+            writeJsonShapeSerializerFunction(writer, model, symbolProvider, shape, memberShape -> true);
```

Comment:
Can replace this predicate with `FunctionalUtils.alwaysTrue()`

Reply:
Cool, this resulted in me just modifying the function signature to take a `Predicate<MemberShape>` rather then a `Function<MemberShape, Boolean>`